### PR TITLE
Cover: Only add filter if needed

### LINKED ISSUE #16233: Blocks: broken with WP 5.3
#16107 may have broken our blocks in older versions of WordPress.

If you test a WordPress 5.3.4 site with Jetpack Master today, you'll run into the following error when trying to insert an Image Compare block for example:

```
TypeError: Object(...) is not a function
    at eval (cover-media-placeholder.js:75)
```

image

@retrofox Do you think you could take a look?

### PR DESCRIPTION
**Pull request by obenland**
Suggestion to fix the `useBlockEditContext` regression for WordPress 5.3.

This emulates the way the VideoPress block is using `useBlockEditContext`, hidden behind an `isSimpleSite` and plan check.

Fixes #16233.

#### Changes proposed in this Pull Request:

* Moves environment checks into `index.js` to only filter the media placeholder when necessary.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Set up a test environment that runs WordPress 5.3.4 with Jetpack master
* Open the editor and try inserting an Image Compare block 
* Make sure the block displays without error

#### Proposed changelog entry for your changes:

* Not needed.

### CODE REVIEW

**Comment by retrofox** [comment]
I haven't tested yet but I have two concerns that maybe don't make sense.

1 - Hooking a function to a filter into a conditional block. I prefer to hook all functions and deal with conditionals into the function. Maybe it's something unfounded, though.
2 - We still are importing the hook in the Media placeholder extension file. Not a big deal neither. Maybe we should add a conditional there.

I think they aren't blockers, though.
Just FYI I've been [working in a different approach](https://github.com/Automattic/jetpack/pull/16240) with fits better (really IHMO) since it continues [this approach](https://github.com/Automattic/jetpack/pull/16171) with the usage of context, getting rid of the hook, and propagating the block name through of the context provider/consumer in a similar way that we propagate the uploading handler.

Aside note: we will need to continue handling the uploading files from other points of the app such as from the `Replace` control button. Context will be the key here as well.

File: extensions/shared/blocks/cover/cover-media-placeholder.js [hunk 1]
```diff
@@ -19,23 +18,23 @@ const ALLOWED_MEDIA_TYPES = [ 'image', 'video' ];
...
+		if ( 'core/cover' !== name ) {
 			return <CoreMediaPlaceholder { ...props } />;
 		}
 
 		const { onError } = props;
 		return (
 			<div className="jetpack-cover-media-placeholder">
 				<CoverMediaContext.Consumer>
-					{ ( onFilesUpload ) => (
+					{ onFilesUpload => (
```
Comment (obenland) [inline extensions/shared/blocks/cover/cover-media-placeholder.js:29]: This and the next two changes seem to have been automated?

**Comment by obenland** [comment]
> We still are importing the hook in the Media placeholder extension file. Not a big deal neither. Maybe we should add a conditional there.

That should be fine as long as it doesn't get called, which this PR makes sure of by moving the `isUpgradeable()` conditional ahead of when `useBlockEditContext` gets called.

File: extensions/shared/blocks/cover/cover-media-placeholder.js (hunk 1 above)
Comment (retrofox) [reply extensions/shared/blocks/cover/cover-media-placeholder.js:29]: yes, I think so

**Comment by retrofox** [comment]
> 1 - Hooking a function to a filter into a conditional block. I prefer to hook all functions and deal with conditionals into the function. Maybe it's something unfounded, though.

Clarifying: hooking a function to a filter into a conditional block, which is also handled by a function that is hooked to another filter. Nesting filters/hooks is my concern.

**Comment by marekhrabe** [comment]
> Hooking a function to a filter into a conditional block. I prefer to hook all functions and deal with conditionals into the function. Maybe it's something unfounded, though.

Personally, I always prefer it the other way (but only if the condition is known beforehand). Otherwise, we are adding a function call and repeated check that will never turn useful.

**Comment by obenland** [comment]
This rebase went haywire, closing in favor of #16271.

**Comment by obenland** [comment]
r209538-wpcom

**Comment by retrofox** [comment]
> > 1 - Hooking a function to a filter into a conditional block. I prefer to hook all functions and deal with conditionals into the function. Maybe it's something unfounded, though.
> 
> Clarifying: hooking a function to a filter into a conditional block, which is also handled by a function that is hooked to another filter. Nesting filters/hooks is my concern.

I meant to this :-(

image

**Pull request closed without merging**